rules/__tests__/prefer-spy-on.test.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const RuleTester = require('eslint').RuleTester;
+const rule = require('../prefer-spy-on');
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 6,
+  },
+});
+
+ruleTester.run('prefer-spy-on', rule, {
+  valid: [
+    'Date.now = () => 10',
+    'window.fetch = jest.fn',
+    'obj.mock = jest.something()',
+    'const mock = jest.fn()',
+    'mock = jest.fn()',
+    'const mockObj = { mock: jest.fn() }',
+    'mockObj = { mock: jest.fn() }',
+  ],
+  invalid: [
+    {
+      code: 'obj.a = jest.fn(); const test = 10;',
+      errors: [
+        {
+          message: 'Use jest.spyOn() instead.',
+          type: 'AssignmentExpression',
+        },
+      ],
+      output: "jest.spyOn(obj, 'a'); const test = 10;",
+    },
+    {
+      code: "Date['now'] = jest['fn']()",
+      errors: [
+        {
+          message: 'Use jest.spyOn() instead.',
+          type: 'AssignmentExpression',
+        },
+      ],
+      output: "jest.spyOn(Date, 'now')",
+    },
+    {
+      code: "obj['prop' + 1] = jest['fn']()",
```

Comment:
could you add a test using template strings as well?

Reply:
I added a test for `window[`${name}`] = jest.fn()`.  `window.name = jest[`fn`]()` doesn't trigger a warning though because i'm using `util.getNodeName` which ignores TemplateLiterals.

Another test that doesn't handle TemplateLiterals is **no-identical-title**. This doesn't trigger warning:
javascript```
test("name", () =>{})
test(`name`, () => {})
```

Should they though?

Reply:
Yeah, I think they should

Reply:
I added a TemplateLiteral case to `util.getNodeName`. This also adds TemplateLiteral support for some other rules which i could add tests for. But maybe in another pull request?

Reply:
Separate PR fleshing out other tests would be great!